main/ui/src/main/java/org/cryptomator/ui/fxapp/FxApplicationModule.java

```diff
@@ -34,17 +36,17 @@
 	}
 
 	@Provides
-	@Named("windowIcon")
+	@Named("windowIcons")
 	@FxApplicationScoped
-	static Optional<Image> provideWindowIcon() {
+	static List<Image> provideWindowIcons() {
 		if (SystemUtils.IS_OS_MAC) {
-			return Optional.empty();
-		}
-		try (InputStream in = FxApplicationModule.class.getResourceAsStream("/window_icon_32.png")) { // TODO: use some higher res depending on display?
-			return Optional.of(new Image(in));
-		} catch (IOException e) {
-			return Optional.empty();
+			return new ArrayList<>();
 		}
+
+		ArrayList<Image> icons = new ArrayList<>();
+		icons.add(new Image(FxApplicationModule.class.getResourceAsStream("/window_icon_32.png")));
```

Comment:
The InputStream doesn't get closed.

Maybe use a helper method for it:

```java
private Image createImageFromResource(String resourceName) throws IOException {
    try (InputStream in = FxApplicationModule.class.getResourceAsStream(resourceName)) {
        return new Image(in);
    }
}
```

And inside of `provideWindowIcons()` simply do

```java
try {
    return List.of( //
        createImageFromResource("/window_icon_32.png"), //
        createImageFromResource("/window_icon_512.png") //
    );
} catch (IOException e) {
    return Collections.emptyList();
}
```

Reply:
The try tricks are really magic to me, revised accordingly, thanks a lot!